ts/screens/modal/UpdateAppModal.tsx

```diff
@@ -45,130 +46,105 @@ const styles = StyleSheet.create({
-        <React.Fragment>
-          <Button block={true} primary={true} onPress={this.openAppStore}>
-            <Text>{I18n.t("btnUpdateApp")}</Text>
-          </Button>
-          <View spacer={true} />
-        </React.Fragment>
-      </View>
-    );
-  }
+  return (
+    <FooterWithButtons
+      type="TwoButtonsInlineThird"
+      leftButton={cancelButtonProps}
+      rightButton={updateButtonProps}
+    />
+  );
+};
 
-  /**
-   * Footer Android buttons
-   */
-  private renderAndroidFooter() {
-    const cancelButtonProps = {
-      cancel: true,
-      block: true,
-      onPress: () => BackHandler.exitApp(),
-      title: I18n.t("global.buttons.close")
-    };
-    const updateButtonProps = {
-      block: true,
-      primary: true,
-      onPress: this.openAppStore,
-      title: I18n.t("btnUpdateApp")
-    };
-
-    return (
-      <FooterWithButtons
-        type="TwoButtonsInlineThird"
-        leftButton={cancelButtonProps}
-        rightButton={updateButtonProps}
-      />
-    );
-  }
+const UpdateAppModal: React.FC = () => {
+  // Disable Android back button
+  const handleBackPress = () => true;
 
-  // Different footer according to OS
-  get footer() {
-    return Platform.select({
-      ios: this.renderIosFooter(),
-      android: this.renderAndroidFooter()
-    });
-  }
+  useEffect(() => {
+    BackHandler.addEventListener("hardwareBackPress", handleBackPress);
 
-  public render() {
-    // Current version not supported
-    return (
-      <Modal>
-        <BaseScreenComponent
-          appLogo={true}
-          goBack={false}
-          accessibilityEvents={{ avoidNavigationEventsUsage: true }}
-        >
-          <Container>
-            <View style={styles.container}>
-              <H2>{I18n.t("titleUpdateApp")}</H2>
-              <Text style={styles.text}>{I18n.t("messageUpdateApp")}</Text>
-              <Image
-                style={styles.img}
-                source={require("../../../img/icons/update-icon.png")}
-              />
-              {this.state.hasError && (
-                <Text style={styles.textDanger}>
-                  {I18n.t("msgErrorUpdateApp")}
-                </Text>
-              )}
-            </View>
-          </Container>
-        </BaseScreenComponent>
-        {this.footer}
-      </Modal>
-    );
-  }
-}
+    return () =>
+      BackHandler.removeEventListener("hardwareBackPress", handleBackPress);
+  }, []);
+
+  // Reset the error state after a given timeout
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    if (error) {
+      setTimeout(() => setError(false), ERROR_MESSAGE_TIMEOUT);
+    }
+  }, [error]);
+
+  // Tries to open the native app store, falling to browser web store
+  const openAppStore = useCallback(async () => {
+    try {
+      await Linking.openURL(storeUrl);
+    } catch (e) {
+      try {
+        await Linking.openURL(webStoreURL);
+      } catch (e) {
+        setError(true);
+      }
+    }
+  }, [setError]);
```

Comment:
imho `useCallback` in this case maybe not so useful, since we are passing `setError` function as a dependency this would never be regenerated and anyway here there's no need to memoize the function. 

I would suggest a more simple implementation of the function as follows:
```typescript
const openAppStore = async () =>
    Linking.openURL(storeUrl).catch(() => {
      openWebUrl(webStoreURL, () => setError(true));
    });
```

you can find `openWebUrl` implementation [here](https://github.com/pagopa/io-app/blob/3f4029c97879ab25ce3318c3d8cc34b95fe98ea9/ts%2Futils%2Furl.ts#L87)

Reply:
Yes, setters are guaranteed to be always the same, but shouldn't `useCallback` prevent the function from being recreated? `setError` is related to the component so I think it should be just removed from the dep. list